airflow/providers/google/cloud/operators/dataflow.py

```diff
@@ -324,6 +344,23 @@ class DataflowTemplatedJobStartOperator(BaseOperator):
             `https://cloud.google.com/dataflow/pipelines/specifying-exec-params
             <https://cloud.google.com/dataflow/docs/reference/rest/v1b3/RuntimeEnvironment>`__
     :type environment: Optional[dict]
+    :param wait_until_finished: (Optional)
+        If True, wait for the end of pipeline execution before exiting. If False,
+        it only waits for it to starts (``JOB_STATE_RUNNING``).
+
+        The default behavior depends on the type of pipeline:
+
+        * for the streaming pipeline, wait for jobs to start,
+        * for the batch pipeline, wait for the jobs to complete.
+
+        .. warning::
+
+            You cannot call ``PipelineResult.wait_until_finish`` method in your pipeline code for the operator
+            to work properly. i. e. you must use asynchronous execution. Otherwise, your pipeline will
+            always wait until finished. For more information, look at:
```

Comment:
By "always wait until finished," do you mean it will block forever?

Why does this happen?

Reply:
`always wait until finished` means that task will wait for the terminal state of the job. It is the current behaviour for batch jobs and it is kept as default behaviour for backward compatibility. But there may be a case that user doesn't want to wait (after successful job start) for the end of the batch job but go further in the DAG.